InfoPopover: improve a11y, fix ESLint warnings, prepare for RootChild modernization

jsnajdr opened the pull request:
Several improvements for the `InfoPopover` component:

<img width="345" alt="Screenshot 2019-05-09 at 11 14 29" src="https://user-images.githubusercontent.com/664258/57442152-a5b92a00-724b-11e9-9f7f-25ff22eaad73.png">

**Prepare for `RootChild` modernization**
Move the popover modal markup outside the clickable "info" element. Otherwise, clicks inside the popover would bubble up to the "info" element and would toggle it (because React Portal behaves that way). One small step towards making #32899 shippable.

**Improve accessibility**
The "info" element has a "button" role, is tabbable and focusable, focus is visible during keyboard navigation, and the popover can be toggled with keyboard. Enter and Space toggles it, ArrowDown opens it and Escape closes it.

There are plenty of other possible a11y improvements we could make: put `aria-label` on the info icon, link the icon and the popover with the `aria-controls` attribute... A11y is such a vast topic 😮 But that's for other PRs. My main goal here was the `RootChild` compatibility.

**Fix ESlint warnings**
Convert React string refs to modern `React.createRef`.

jsnajdr committed:
InfoPopover: improve a11y, fix ESLint warnings, prepare for RootChild modernization

flootr commented on client/components/info-popover/index.jsx:
```diff
@@ -68,30 +87,39 @@ export default class InfoPopover extends Component {
-					rootClassName={ this.props.rootClassName }
+			<Fragment>
+				<span
+					role="button"
```
❓ I guess we could also use a `button` element instead of assigning `role=button` to `span`? I'm not an accessibility expert, just curious - not crucial for this PR.

flootr reviewed (approved):
as opposed to the current state on master this works very well for me except for the `ArrowDown` trigger - i'm not exactly sure why though. feel free to ignore this if it works for you. thank you 👍

jsnajdr committed:
Use button tag for the popover button

jsnajdr commented:
@flootr I addressed your review comments:
- I changed the `span` into `button` and it works great! I don't know why I didn't do that before 😄 Controlling by keyboard works out of the box, no need for extra keydown listeners
- the ArrowDown even indeed doesn't work. I'm sure it worked when I tested it few weeks ago, but it doesn't now ☹️ Anyway, we delegate all the keyboard events to the browser now anyway, using the default native behavior of the `button` tag.

jsnajdr committed:
Set button type to button (default is submit). Prevents submitting parent forms
